### DISCUSSION TITLE
Home Assistant Error message: Installed add-on has been removed from repository

### DISCUSSION THREAD

**Discussion by MSoegtrop** [OpenCCU/OpenCCU#2841] [🇺🇸 Q&A] [answered]
### Describe the issue you are experiencing

After updating to the latest version of the HomaAssiatnt RaspberryMatic plugin for HomeAssistant I got this error message this morning:
```
Add-on RaspberryMatic CCU has been removed from the repository it was installed from. This means it will not get updates, and backups may not be restored correctly as the supervisor may not be able to build/download the resources required.

Clicking submit will uninstall this deprecated add-on. Alternatively, you can check [Home Assistant help](https://www.home-assistant.io/help/) and the [community forum](https://community.home-assistant.io/) for alternatives to migrate to.
```
I darkly remember that you mentioned before that the HomeAssistant plugin for RaspberryMatic is maintained by someone else and that I should report related issues somewhere else, but the link in HomeAssistant points here.

### Describe the behavior you expected

No error message

### Steps to reproduce the issue

Update a RaspberryMatic installation in HomeAssistant to version 3.77.7.20240826 and reboot the system.

### What is the version this bug report is based on?

3.77.7.20240826

### Which base platform are you running?

rpi3 (RaspberryPi3, ARM64/aarch64)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
Please give details what logs you would need.
```

### Additional information

I am running raspberryMatic on HomeAssistant on an original CCU3.

**Comment by jens-maus** [comment]
Please show a screenshot. However, this might simply happen in case GitHub connection might get interrupted for some time. The "RaspberryMatic CCU" Add-on is definitely hosted by this repository here and by nobody else, so no other idea where this might come from.

**Comment by MSoegtrop** [comment]
I attached a screen shot. The issue seems to be persistent for > 11 hours. My initial setup was 2 years ago. Is it possible that URLs changed meanwhile and I have an old URL?

![Screenshot add-on removed](https://github.com/user-attachments/assets/37db3888-909c-4333-beab-fbe24d5863ca)

**Comment by jens-maus** [comment]
Then also please show the respository URL you used.

**Comment by MSoegtrop** [comment]
The link in Home Assistant plugin information page goes here (https://github.com/jens-maus/RaspberryMatic/) - not sure if there are any other hidden links.

The plugin info page has btw. the URL: http://CCU3-IP:8123/hassio/addon/de838cd8_raspberrymatic/info

**Comment by MSoegtrop** [comment]
I checked my setup notes. Apparently I did re-setup everything last year on Oct 22nd 2023 following these instructions:

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant
https://github.com/jens-maus/RaspberryMatic/wiki/HomeAssistant-Integration

with these deviation notes (which shouldn't have anything to do with this but you might find them interesting anyway):

- Bluetooth loast after installing HACS
  - The moment I installe HACS, bluetooth was lost and did not come up again
    - a full system reboot fixed this

- Enable GPIO access for RF board
  - The boot drive does not seem to be accessible via ssh
    - best way seems to be to put in SD card into a Windows PC and edit config on the 32MB FAT boot drive

**Comment by Baxxy13** [comment]
Check: Add-ons --> ADD-ON STORE (lower right) --> Repositorys (3-Dot Menu upper right)...
Should look like:
![HA_AddOn-Repositorys](https://github.com/user-attachments/assets/aeb16c51-0822-4320-9c89-7bfd3d599dd3)

**Comment by MSoegtropIMC** [comment]
For me this looks as below. The URL is the same, but it says it is invalid. Another github repo seems to work. The error message was renewed after a full reboot. Also I want to note again that this error message appeared at a reboot after updating RaspbarryMatic to 3.77.7.20240826.

![Screenshot add-on removed 2](https://github.com/user-attachments/assets/1c20fae7-2971-4e03-9b07-d0437e4884d2)

Internet connectivity is fine btw - I have the check enabled in RaspberryMatic and not any error reports.

**Comment by MSoegtropIMC** [comment]
@jens-maus : do you have an idea what could be wrong?

**Comment by jens-maus** [comment]
I would suggest you take a backup within RaspberryMatic, then remove the whole addon and the repositor and then readd everything again...

**Comment by MSoegtropIMC** [comment]
@jens-maus : There is some discussion here:

https://github.com/home-assistant/core/issues/118917

it is a bit confusing but one can interpret it such that in June 2024 there was a change in HomeAssistant which requires to put additional information on the github repos for plugins. Or that this requirement is actually older but since June 2024 one gets the above error message discussed here in case the information is not there. Quite a few plugins seem to be affected by this.

Can you please have a look?

**Reply by jens-maus** [reply]
Thanks for the links and the additional information. However, after having checked these links and also after having tried it myself again I still think that this must be a local issue in your own installation only and not a general issue in RaspberryMatic providing its HA add-on repository.

So please try the following procedure to solve the issue for you:

1. Login into RaspberryMatic WebUI.
2. Create a sbk-Backup and download it to your local PC system.
3. Stop the RaspberryMatic HA-addon and uninstall it completely.
4. Remove the RaspberryMatic Add-on store in the HA configuration
5. Reboot your HA instance
6. Re-add the RaspberryMatic Add-on Store and re-install the HA-addon accordingly.
7. Login to the new/fresh RaspberryMatic WebUI
8. Restore the sbk-backup within the WebUI.

I think this should solve the issue which IMHO is a minor issue/bug in HA itself where it somehow happend that in your installation HA thinks the RaspberryMatic Add-on repository vanished, which of course is not the case.

**Comment by MSoegtropIMC** [comment]
@jens-maus : thank you for the detailed instructions!

I am currently reconsidering if it is a good idea to install RaspberryMatic inside HomeAssistant. Just because one can do something doesn't mean one should. My reasoning:

- some of the functionality I wanted to use from HomeAssistant does not really work as intended and I have to rewrite this any way
- I am not using that much functionality from HomeAssistant not available on RaspberryMatic as well (one thing are Fritz actors).
- I might be better off writing this as RaspberryMatic plugin right away since the overall system would be much simpler this way
- Running HomeMatic + Rapsberry Matic on a CCU3 works but is a bit of a stretch

One thing I am working on is a proper integration of SMA solar systems.

In case I follow your instructions I would clone the SD card before and check afterwards what the difference is.

**Comment by henkha** [comment]
I am bumping this post up, because just yesterday, i ran into exactly the same problem on my home assistant installation.

![image](https://github.com/user-attachments/assets/ce853c37-9a40-4b21-bfdd-9d12bf425db0)

![image](https://github.com/user-attachments/assets/71fc2aab-1c49-4639-9f35-17394b494bf2)

![image](https://github.com/user-attachments/assets/0f4dada8-5450-46a6-bfba-c5accc3db7f8)

![image](https://github.com/user-attachments/assets/a5041d44-6472-4faa-b7d2-bf29cb5fe832)

I will go and try now to take a backup within RaspberrymaticCCU and reinstall the AddOn from scratch.

**Comment by Baxxy13** [comment]
Perhaps you are all affected by this:
https://github.com/home-assistant/supervisor/issues/5699

**Reply by jens-maus** [reply]
I also think this is clearly a bug/issue in HomeAssistant itself and not particular related to the RaspberryMatic Add-on. Thought, never faced this issue myself, but I think depending on the GitHub/Internet-connectivity this could easily happen and have to be addressed in HA itself in case an add-on repository is sometimes not reachable.

**Comment by Pa7rickStar** [comment]
I just got the same error message today (HA Supervisor & Core are up to date). The repository url I'm using is: https://github.com/OpenCCU/OpenCCU/

**Comment by jens-maus** [comment, marked as answer]
See here for the official guide on porting from RaspberryMatic HA Addon to OpenCCU HA Addon:
https://github.com/OpenCCU/OpenCCU/wiki/Installation#migration-des-raspberrymatic-ha-add-ons

**Reply by Pa7rickStar** [reply]
Thanks! I stumbled across this myself. 🙈

Migration did not work at first because I couldn't restore the rasperrymatic backup in the openccu instance (when clicking restore in the GUI nothing happened; the Home Assistant OpenCCU add-on showed an nginx error that magically disappeared after 5 minutes).

If anyone has the same issue: Using cli to restore the backup worked for me. I copied the rasperrymatic backup from my machine to the Home Assistant config folder using smb; ssh into Home Assistant and used scp to copy the backup into the OpenCCU add-on container. ssh into the ccu and did the restore.
